src/main/io/gps.c

```diff
@@ -521,50 +691,155 @@ static void ubloxSendPollMessage(uint8_t msg_id)
+        offset += ubloxAddValSet(&tx_buffer, CFG_NAVSPG_OUTFIL_PDOP, payload, offset); // 42
+
+        payload[0] = (uint8_t)(250 >> (8 * 0));
+        payload[1] = (uint8_t)(250 >> (8 * 1));
+        offset += ubloxAddValSet(&tx_buffer, CFG_NAVSPG_OUTFIL_TDOP, payload, offset); // 48
+
+        payload[0] = (uint8_t)(100 >> (8 * 0));
+        payload[1] = (uint8_t)(100 >> (8 * 1));
+        offset += ubloxAddValSet(&tx_buffer, CFG_NAVSPG_OUTFIL_PACC, payload, offset); // 54
+
+        payload[0] = (uint8_t)(300 >> (8 * 0));
+        payload[1] = (uint8_t)(300 >> (8 * 1));
+        offset += ubloxAddValSet(&tx_buffer, CFG_NAVSPG_OUTFIL_TACC, payload, offset); // 60
+
+        payload[0] = (uint8_t)(0 >> (8 * 0));
+        offset += ubloxAddValSet(&tx_buffer, CFG_MOT_GNSSSPEED_THRS, payload, offset); // 65
+
+        payload[0] = (uint8_t)(200 >> (8 * 0));
+        payload[0] = (uint8_t)(200 >> (8 * 1));
+        offset += ubloxAddValSet(&tx_buffer, CFG_MOT_GNSSDIST_THRS, payload, offset); // 71
+
+        payload[0] = (uint8_t)(60 >> (8 * 0));
+        offset += ubloxAddValSet(&tx_buffer, CFG_NAVSPG_CONSTR_DGNSSTO, payload, offset); // 76
+
+        payload[0] = (uint8_t)(0 >> (8 * 0));
+        offset += ubloxAddValSet(&tx_buffer, CFG_NAVSPG_INFIL_NCNOTHRS, payload, offset); // 81
+
+        payload[0] = (uint8_t)(0 >> (8 * 0));
+        offset += ubloxAddValSet(&tx_buffer, CFG_NAVSPG_INFIL_CNOTHRS, payload, offset); // 86
+
+        ubloxSendConfigMessage(&tx_buffer, MSG_CFG_VALSET, offsetof(ubxCfgValSet_t, cfgData) + offset, true);
+    } else {
+        memset(&tx_buffer, 0, sizeof(ubxMessage_t));
+        tx_buffer.payload.cfg_nav5.mask = 0xFFFF;
+        tx_buffer.payload.cfg_nav5.dynModel = model == 0 ? 0 : model + 1; //no model with value 1
+        tx_buffer.payload.cfg_nav5.fixMode = 3;
+        tx_buffer.payload.cfg_nav5.fixedAlt = 0;
+        tx_buffer.payload.cfg_nav5.fixedAltVar = 10000;
+        tx_buffer.payload.cfg_nav5.minElev = 5;
+        tx_buffer.payload.cfg_nav5.drLimit = 0;
+        tx_buffer.payload.cfg_nav5.pDOP = 250;
+        tx_buffer.payload.cfg_nav5.tDOP = 250;
+        tx_buffer.payload.cfg_nav5.pAcc = 100;
+        tx_buffer.payload.cfg_nav5.tAcc = 300;
+        tx_buffer.payload.cfg_nav5.staticHoldThresh = 0;
+        tx_buffer.payload.cfg_nav5.dgnssTimeout = 60;
+        tx_buffer.payload.cfg_nav5.cnoThreshNumSVs = 0;
+        tx_buffer.payload.cfg_nav5.cnoThresh = 0;
+        tx_buffer.payload.cfg_nav5.staticHoldMaxDist = 200;
+        tx_buffer.payload.cfg_nav5.utcStandard = ubloxUTCStandardConfig_int[gpsConfig()->gps_ublox_utc_standard];
+
+        ubloxSendConfigMessage(&tx_buffer, MSG_CFG_NAV_SETTINGS, sizeof(ubxCfgNav5_t), false);
+    }
 }
 
-static void ubloxSendNAV5Message(uint8_t model)
+static void ubloxSendNavX5Message(void) {
+    ubxMessage_t tx_buffer;
+
+    if (gpsData.ubloxM9orAbove) {
+        uint8_t payload[1];
+        payload[0] = (uint8_t)(1 >> (8 * 0));
```

Comment:
This makes little sense here, as you are not loading more than a single `1` into the payload
```suggestion
        uint8_t payload[1];
        payload[0] = 1;
```